cEP-0012.md

```diff
@@ -0,0 +1,152 @@
+
+### Set 'Do nothing' as the default action
+
+If a user chooses a non-existent action (or the input is wrong) just exit from the program with 'Do nothing'. This is for reuse of 'Do nothing'.
+
+The section below describes how we should make these changes.
+
+How
+---
+
+### Remove most of the text from the action selection screen
+
+Most of the text that should be changed is in the 'ConsoleInteraction.py' file. Do changes based on the feedback gathered with
+Google Form.
+
+Make changes in the 'ConsoleInteraction.py' based on the feedback gathered with Google Form.
+
+### Use letters instead of numbers for selecting actions
+
+Usage of letters would make usage of coala faster, as one could just remember
+that `a` stands for 'apply patch' for instance, and they would not have to check
+the list for every result to see what number 'apply patch' has been assigned.
+
+Do masive changes in 'result_actions' directory where are kept all the actions.
+
+### Allow chaining actions in a single input
+
+Mockup:
+
+[    ] Do (N)othing
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] Choose an action:
+
+If a user will press 'C':
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
+
+Any combination of letters (upper/lower) will be accepted (e.g: a-I-N).
+If one of the actions requested isn't possible, a warning will be printed (after all possible actions are applied) with the letter that couldn't be satisfied.
+
+1. Create a new file 'ChainAction.py' and a new class 'ChainAction' in 'result_actions' directory
+2. Implement the logic in the new class
+3. Parse corectly the input
+3. Use the new class in 'ConsoleInteraction'
+
+### Add a 'select action for all results' option
+
+Mockup:
+
+[    ] Do (N)othing
+[    ] (O)pen file(s)
+[    ] (A)pply the patch
+[    ] Add (i)gnore comment
+[    ] (C)hain actions
+[    ] (S)elect action for all results
+[    ] Choose an action:
+
+If the user will press 'S':
+
+==== Apply an action for all results ====
+[    ] Do (N)othing
+[    ] (O)pen file
+[    ] (A)pply the patch
+[    ] Add (I)gnore comment
+[    ] Choose an action:
+
+So, let's say that a user wants to add an ignore comment in all files.
+The user should choose '(S)elect action for all results' and after choose the 'i' option.
+This will add an ignore comment in all of the files used in coafile/command without asking another action.
+
+### Reprompt after running actions
+
+This should work if the user chooses to (C)hain actions.
+
+[    ] Enter the letters of the patches you want to apply ('-' between any letter): a-i-n
+[    ] Reprompt after running actions (Y/N)?
+
+If the user chooses 'Y', after all the actions are applied, the interface will be reprompted with the possible actions.
+
+[    ] Do (N)othing
```

Comment:
Please use the multi-line code blocks for the mock-ups. It looks weird in the "Rich text view"